### ISSUE TITLE
While renaming a file, the cursor cannot be moved using the cursor keys

### ISSUE THREAD

**Issue by nesshoever** [Microsoft/vscode#39309]
- VSCode Version: Code - Insiders 1.19.0-insider (ddb27c63122c09a3c4800cc4d928cd9faf97cef1, 2017-11-29T06:14:43.592Z)
- OS Version: Linux x64 4.10.0-40-generic
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
ng-template|Ang|0.1.7
Angular-BeastCode|Mik|5.1.3
vscode-docker|Pet|0.0.22
path-intellisense|chr|1.4.2
tslint|eg2|1.0.24
vscode-npm-script|eg2|0.3.3
json-tools|eri|1.0.2
csharpextensions|jch|1.3.0
vscode-nuget-package-manager|jmr|1.1.5
csharp|ms-|1.13.1
mono-debug|ms-|0.15.8
PowerShell|ms-|1.5.1
debugger-for-chrome|msj|3.5.0
syncing|non|1.4.9
vscode-icons|rob|7.18.1
yo|sam|0.9.3
autoimport|ste|1.5.3
gitblame|wad|2.2.0

---

Steps to Reproduce:

1. Open a project
2. Select a file
3. Hit F2 to rename a file
4. Try to move the cursor in the selected filename -> will fail: instead the selection in the file tree is moved

Reproduces without extensions: Yes

**Comment by Narkoleptika** [comment]
I'm also experiencing this issue. It seems that the focus is on the file explorer rather than the file name field.

In addition to not being able to move the cursor, I can't use the A, L or backspace keys. I'm sure there are more, but those are the main ones I've run into.

For now, my workaround is to type out the file name in a new untitled tab and then right click and paste on the file name field.

**Comment by Tetheta** [comment]
Also having this issue, though for me I can't use the delete key, backspace works fine.
Even stranger, if my VSCodeVIM plugin is enabled, I can use j/k and enter to navigate the tree, but then I can't type j or k when renaming a file, it runs around the tree instead.

Here's my info:
- VSCode Version: Code - Insiders 1.19.0-insider (6e90f8995da4bd999e8983b8e480a34045c1dfcf, 2017-11-29T05:14:15.979Z)
- OS Version: Windows_NT x64 10.0.15063
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
rainbow-brackets|2gu|0.0.6
project-manager|ale|0.23.0
npm-intellisense|chr|1.3.0
bracket-pair-colorizer|Coe|0.10.14
vscode-markdownlint|Dav|0.12.0
vscode-eslint|dba|1.4.3
gitlens|eam|6.2.0
vsc-material-theme|Equ|1.1.2
prettier-vscode|esb|0.26.0
latex-workshop|Jam|3.7.0
node-debug2|ms-|1.19.0
debugger-for-chrome|msj|3.5.0
indent-rainbow|ode|0.7.2
vscode-versionlens|pfl|0.19.1
material-icon-theme|PKi|3.1.0
java|red|0.14.0
vim|vsc|0.10.5
vscode-react-native|vsm|0.5.7
vscode-import-cost|wix|2.5.1
html-css-class-completion|Zig|1.14.0

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

I suggested in #21952 a change, similar to what happens in Eclipse.

But they closed the case claiming not to do in the next 6-12 months.

I think this case here, and one more example why not put the style I spoke in case #21952 , to solve this problems, and when I need to copy a file that will have almost the same functionality as another, replacing only the name.

//ping @isidorn 🤔
